src/converters.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import ViewContainerElement from '../engine/view/containerelement.js';
+import ViewEmptyElement from '../engine/view/emptyelement.js';
+import ModelElement from '../engine/model/element.js';
+import { isImageWidget } from './utils.js';
+
+/**
+ * Returns function that converts image view representation:
+ *
+ *		<figure class="image"><img src="..." alt="..."></img></figure>
+ *
+ * to model representation:
+ *
+ *		<image src="..." alt="..."></image>
+ *
+ * @returns {Function}
+ */
+export function viewToModelImage() {
+	return ( evt, data, consumable, conversionApi ) => {
+		const viewFigureElement = data.input;
+
+		// *** Step 1: Validate conversion.
+		// Check if figure element can be consumed.
+		if ( !consumable.test( viewFigureElement, { name: true, class: 'image' } ) ) {
+			return;
+		}
+
+		// Check if image element can be converted in current context.
+		if ( !conversionApi.schema.check( { name: 'image', inside: data.context } ) ) {
+			return;
+		}
+
+		// Check if img element is placed inside figure element and can be consumed with `src` attribute.
+		const viewImg = viewFigureElement.getChild( 0 );
+
+		if ( !viewImg || viewImg.name != 'img' || !consumable.test( viewImg, { name: true, attribute: 'src' } ) ) {
+			return;
+		}
+
+		// *** Step2: Convert to model.
+		consumable.consume( viewFigureElement, { name: true, class: 'image' } );
+		consumable.consume( viewImg, { name: true, attribute: 'src' } );
+
+		// Create model element.
+		const modelImage = new ModelElement( 'image', {
+			src: viewImg.getAttribute( 'src' )
+		} );
+
+		// Convert `alt` attribute if present.
+		if ( consumable.consume( viewImg, { attribute: [ 'alt' ] } ) ) {
+			modelImage.setAttribute( 'alt', viewImg.getAttribute( 'alt' ) );
+		}
+
+		data.output = modelImage;
+	};
+}
+
+/**
+ * Returns model to view selection converter. This converter is applied after default selection conversion is made.
+ * It creates fake view selection when {@link engine.view.Selection#getSelectedElement} returns instance of image widget.
+ *
+ * @param {Function} t {@link utils.Locale#t Locale#t function} used to translate default fake selection's label.
+ * @returns {Function}
+ */
+export function modelToViewSelection( t ) {
```

Comment:
There should be no special selection converter for image widget. It should be general to widgets. So that label must be handled differently.